contrib/kube-prometheus/README.md

```diff
@@ -117,26 +130,24 @@ jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {}
 
 ```
 
-> Note you need `jsonnet` and `gojsonyaml` (`go get github.com/brancz/gojsontoyaml`) installed. If you just want json output, not yaml, then you can skip the pipe and everything afterwards.
-
-This script reads each key of the generated json and uses that as the file name, and writes the value of that key to that file.
-
-> You can also run this script executing the command `make generate-raw` from kube-prometheus base directory of this repository but the above option it is recommended so that you run it in your own infrastructure repository.
+This script runs the jsonnet code, then reads each key of the generated json and uses that as the file name, and writes the value of that key to that file, and converts each json manifest to yaml.
 
 ## Configuration
 
-A hidden `_config` field is located at the top level of the object this library provides. These are the available fields with their respective default values:
+Jsonnet has the concept of hidden fields. These are fields, that are not going to be rendered in a result. This is used to configure the kube-prometheus components in jsonnet. In the example jsonnet code of the above [Usage section](#Usage), you can see an example of this, where the `namespace` is being configured to be `monitoring`. In order to not override the whole object, use the `+::` construct of jsonnet, to merge objects, this way you can override individual settings, but retain all other settings and defaults.
+
+These are the available fields with their respective default values:
 
 ```
 {
 	_config+:: {
         namespace: "default",
```

Comment:
some places you recommend installing prometheus in namespace "monitoring".. and here you use "default".. a bit confusing as to what is the recommendation. What do you do ?